sys/include/riot_hdr.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_riot_hdr RIOT header helpers and tools
+ * @ingroup     sys
+ * @{
+ *
+ * The header contains
+ *
+ * - "RIOT" as magic number
+ * - the application version
+ * - the address where to find the RIOT firmware
+ * - the checksum of the three previous fields
+ *
+ * @file
+ * @brief       RIOT "partition" header and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#ifndef RIOT_HDR_H
+#define RIOT_HDR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+/**
+ * @brief  Magic number for riot_hdr
+ *
+ */
+#define RIOT_HDR_MAGIC         0x544f4952 /* "RIOT" */
+
+/**
+ * @brief Structure to store image header - All members are little endian
+ * @{
+ */
+typedef struct {
+    uint32_t magic_number;     /**< header magic_number (always "RIOT")              */
+    uint32_t version;          /**< Integer representing the partition version       */
+    uint32_t start_addr;       /**< Address after the allocated space for the header */
+    uint32_t chksum;           /**< checksum of riot_hdr                             */
```

Comment:
Why it does not indicate the length of the image?

Reply:
@jcarrano The header struct here is for a minimal bootloader application. It is however possible to extend this struct for different types of bootloaders. So if the current bootloader does not require it, I'd rather keep it out of the struct.

Reply:
I'd say in the worse case you always have the slot size which can be used for that purpose. Otherwise, while receiving the data allocated to that slot you always get the size, so if you need to perform other operations you can pass that information if needed.

Anyways, the bootloader proposed on #10065 is not ready to perform any swapping, and a big change would be needed for that. When those changes come, modifications to this struct might be done.